testplan/common/utils/comparison.py

```diff
@@ -663,6 +764,26 @@ def _rec_compare(lhs,
         rhs=fmt(rhs))
 
 
+def stringify(x, y):
+    """
+    Convert all objects to strings using str() before making the comparison.
+    """
+    x_ = str(x)
+    y_ = str(y)
+    ret = x_ == y_
+
+    if not ret:
+        if any(isinstance(val, float) or isinstance(val, decimal.Decimal)
```

Comment:
Yes, people is comparing typed FixMsg with untyped Fixmsg all the time, and we need 0.0 to match '0'

Reply:
Because fix messages are stringly typed, and the semantics of tag values may be fundamentally different. 
* 38=0
* 150=0
* 16070=0
* 16103=0

It is hard to tell for which ones should be safely comparable to `'0'`, I think it is right to assume all. 0 may be a char, a boolean flag, a double like price, or a quantity-related internal tag solely used by systems for stocks where qty unit is 1.

Reply:
Renamed this function to untyped_fixtag and updated its docstring